src/Azure/Orleans.Persistence.AzureStorage/Options/AzureTableStorageOptionsFormatterResolver.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.Generic;
+using Microsoft.Extensions.Options;
+using Orleans.Runtime.Configuration;
+
+namespace Orleans.Hosting
+{
+    public class AzureTableStorageOptionsFormatterResolver : IOptionFormatterResolver<AzureTableStorageOptions>
+    {
+        private IOptionsSnapshot<AzureTableStorageOptions> optionsAccessor;
+        public AzureTableStorageOptionsFormatterResolver(IOptionsSnapshot<AzureTableStorageOptions> optionsAccessor)
+        {
+            this.optionsAccessor = optionsAccessor;
+        }
+
+        public IOptionFormatter<AzureTableStorageOptions> Resolve(string name)
+        {
+            return new AzureTableStorageOptionsFormatter(optionsAccessor.Get(name));
+        }
+
+        private class AzureTableStorageOptionsFormatter : IOptionFormatter<AzureTableStorageOptions>
+        {
+            public string Category { get; } = "Storage";
+
+            public string Name => nameof(AzureTableStorageOptions);
+
+            private AzureTableStorageOptions options;
+
+            public AzureTableStorageOptionsFormatter(AzureTableStorageOptions options)
```

Comment:
add `name` to this formatter's constructor, and fill in its `Name` property to include the `name`. Since this is a named formatter, we need some identifier to tell user which named option it is printing.